src/views/citation.ts

```diff
@@ -49,30 +49,29 @@ export class CitationView<PropsType extends BaseNodeProps>
       return
     }
 
+    const id = this.node.attrs.id
     const element = document.createElement('span')
+    element.id = id
     element.classList.add('citation')
     if (this.node.attrs.dataTracked?.length) {
-      element.classList.add(...getChangeClasses(this.node.attrs.dataTracked))
-      element.dataset.trackId = this.node.attrs.dataTracked[0].id
+      const change = this.node.attrs.dataTracked[0]
+      element.setAttribute('data-track-id', change.id)
+      element.setAttribute('data-track-status', change.status)
+      element.setAttribute('data-track-op', change.operation)
+      element.classList.add(...getChangeClasses([change]))
     }
 
-    const text = bib.renderedCitations.get(this.node.attrs.id)
+    const text = bib.renderedCitations.get(id)
     const fragment = sanitize(
       text && text !== '[NO_PRINTED_FORM]' ? text : ' ',
       {
         ALLOWED_TAGS: ['i', 'b', 'span', 'sup', 'sub', '#text'],
       }
     )
     element.appendChild(fragment)
-    this.dom.className = 'citation-wrapper'
     this.dom.innerHTML = ''
     this.dom.appendChild(element)
-    addCommentToLeafNode(
-      this.getPos(),
-      this.getPos() + this.node.nodeSize,
-      this.view.state,
-      this.dom
-    )
```

Comment:
can we render the comment inside the citation dom node? keeping it beside span node will not position the comment at the right position

Reply:
Is doing that with CSS not sufficient?

Reply:
if it works with other CSL style then it is ok, can we please ask QA to recheck case of LEAN-2974 with this task

Reply:
Got it.